src/Avalonia.Controls/DateTimePickers/DatePicker.cs

```diff
@@ -0,0 +1,517 @@
+                SetAndRaise(MaxYearProperty, ref _maxYear, value);
+
+                if (SelectedDate.HasValue && SelectedDate.Value > value)
+                    SelectedDate = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the minimum year for the picker
+        /// </summary>
+        public DateTimeOffset MinYear
+        {
+            get => _minYear;
+            set
+            {
+                if (value > MaxYear)
+                    throw new InvalidOperationException("MinDate cannot be greater than MaxDate");
+                SetAndRaise(MinYearProperty, ref _minYear, value);
+
+                if (SelectedDate.HasValue && SelectedDate.Value < value)
+                    SelectedDate = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the month format
+        /// </summary>
+        public string MonthFormat
+        {
+            get => _monthFormat;
+            set => SetAndRaise(MonthFormatProperty, ref _monthFormat, value);
+        }
+
+        /// <summary>
+        /// Gets or sets whether the month is visible
+        /// </summary>
+        public bool MonthVisible
+        {
+            get => _monthVisible;
+            set
+            {
+                SetAndRaise(MonthVisibleProperty, ref _monthVisible, value);
+                SetGrid();
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the year format
+        /// </summary>
+        public string YearFormat
+        {
+            get => _yearFormat;
+            set => SetAndRaise(YearFormatProperty, ref _yearFormat, value);
+        }
+
+        /// <summary>
+        /// Gets or sets whether the year is visible
+        /// </summary>
+        public bool YearVisible
+        {
+            get => _yearVisible;
+            set
+            {
+                SetAndRaise(YearVisibleProperty, ref _yearVisible, value);
+                SetGrid();
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the Selected Date for the picker, can be null
+        /// </summary>
+        public DateTimeOffset? SelectedDate
+        {
+            get => _selectedDate;
+            set
+            {
+                var old = _selectedDate;
+                SetAndRaise(SelectedDateProperty, ref _selectedDate, value);
+                SetSelectedDateText();
+                OnSelectedDateChanged(this, new DatePickerSelectedValueChangedEventArgs(old, value));
+            }
+        }
+
+        /// <summary>
+        /// Raised when the <see cref="SelectedDate"/> changes
+        /// </summary>
+        public event EventHandler<DatePickerSelectedValueChangedEventArgs> SelectedDateChanged;
+
+
+        protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
+        {
+            _areControlsAvailable = false;
+            base.OnApplyTemplate(e);
+            _flyoutButton = e.NameScope.Find<Button>("FlyoutButton");
```

Comment:
I prefer `PART_` notation

Reply:
>Does UWP not use that convention?
No, UWP/WinUI does not follow the PART_ convention in templates. After using it for a long time I can say I prefer the PART_ naming convention. While it technically violates naming rules, it makes it absolutely clear which parts are important in the template and used in code.

Reply:
I'm probably in the minority here, but I prefer following UWP and not using the PART_ notation.

Reply:
Everyone has their preferences, I'm curious though if there is a reason you like the UWP notation. Is it because it follows naming guidelines and fits in better?